### PR TITLE
Settings: add a caution message for the Jetpack Blocks toggle

### PR DESCRIPTION
**Pull request by samiff**
#### Changes proposed in this Pull Request:

Adds a small caution message for the Jetpack Blocks setting toggle.

#### Jetpack product discussion

Raised during internal 10.8-beta testing: p8oabR-PC-p2#comment-6158

> I wonder if we should add a warning in the notice that pops up when you deactivate the blocks, something along the lines of “Careful, if you’ve used Jetpack blocks in some of your posts and pages in the past, those will stop working properly now”.

This isn't a popup, but is better than nothing for now.

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Post-patch, observe the new caution message when searching for 'blocks' in settings: `/wp-admin/admin.php?page=jetpack#/writing?term=blocks`

Screen Shot 2022-04-01 at 3 12 57 PM

Any suggestions on the phrasing used there?

> Caution: if there are Jetpack blocks used in existing posts or pages, disabling this setting will cause those blocks to stop working.

### CODE REVIEW

**Comment by samiff** [comment]
@jeherve Do you know why we don't show the Jetpack Block settings card if the Classic Editor plugin is active?

https://github.com/Automattic/jetpack/blob/2ef743a8e4a3f4012a24b663020a0b0da2c01dfd/projects/plugins/jetpack/_inc/client/writing/composing.jsx#L225

I was confused when I didn't see the setting on my site initially. The Classic Editor plugin does allow the block editor to be set as the default, or allows switching between the two. Wasn't sure if there was particular reason we make that check.

File: projects/plugins/jetpack/_inc/client/writing/composing.jsx [hunk 1]
```diff
@@ -198,6 +198,12 @@ export class Composing extends React.Component {
 										'jetpack'
 									) }
 								</span>
+								<span className="jp-form-setting-explanation">
```
Comment (jeherve) [inline projects/plugins/jetpack/_inc/client/writing/composing.jsx:201]: What do you think about adding a condition so this only shows when the toggle is active?

Comment (samiff) [reply projects/plugins/jetpack/_inc/client/writing/composing.jsx]: 6c75359b2c2c040f06d33d1519ea2d13081c410f

**Comment by jeherve** [comment]
> The Classic Editor plugin does allow the block editor to be set as the default, or allows switching between the two.

Ah, that's a good point. I added that in, because in my mind the classic editor is all about getting rid of all things blocks, but you're right that it can be more than that. I'll remove that in a follow-up PR. 👍

**Review by jeherve — APPROVED** [review]

**Comment by jeherve** [comment]
>  I'll remove that in a follow-up PR. 👍

#23785.

**Pull request merged**